app/models/update_log.rb

```diff
@@ -4,14 +4,42 @@ def self.setting_record
     Setting.find_or_create_by(key: 'metrics_update')
   end
 
-  def self.log_update(time)
+  def self.log_delay
     setting = self.setting_record
-    setting.value["last_update"] = time
+    constant_update =  setting.value['constant_update']
+    if constant_update 
+      seconds = constant_update.keys.length > 1 ? constant_update.map {|key, value| (Time.parse(constant_update[key]["end_time"].to_s) - Time.parse(constant_update[key-1]["end_time"].to_s) if key != constant_update.keys[0]).to_i} : [0]
+      setting.value['average_delay'] = seconds.inject(:+)/(seconds.length-1) if seconds.length > 1
+    end
     setting.save
   end
 
+  def self.log_updates(starttime, endtime)
```

Comment:
This class will benefit from separating out the high-level concepts of what it is doing, and the low-level details of how it is doing them... along with separating out the methods intended to be public from the methods that are helpers and should be considered private because we only want to call them from other methods within this class.

At the high level, there are basically two entry points: the one for fetching data to display on the frontend, and the one for logging a new update. Those should be at the top of the file.

One strategy I like is to write those public methods completely in terms of private helper methods that tell a clear story of what steps are involved in the process. So for `log_update`, it might be something like:
```ruby
def self.log_update(update_data)
  add_new_log update_data
  delete_oldest_log
  update_average_delay
  save_setting_record
end
```

Reply:
Nice feedback. Would you create that many methods to do so? Or is just the example?

Reply:
Yes, I would create separate methods for each of those steps.